### PR TITLE
Update to Ruby 3.2 and Rails 6.1 [PIMCORE-450]

### PR DESCRIPTION
**Pull request by erikkessler1**
## Problem

We are using versions of Ruby and Rails that are EOL.

Closes #410.

## Changes

This PR upgrades to Ruby 3.2.2 and Rails 6.1.

## Notes

It would be nice to get to Ruby 3.3 and Rails 7+, but I figured this would be a nice stepping stone. Also, I was running into [this issue](https://bugs.ruby-lang.org/issues/20085) with Ruby 3.3.0 when trying to run a containerized version of the application locally, so we might need to wait for Ruby 3.3.1.

prime: @atsheehan 

[PIMCORE-450]

[PIMCORE-450]: https://salsify.atlassian.net/browse/PIMCORE-450?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -31,10 +30,6 @@ group :development do
   gem 'avro_turf', '>= 0.8.0', require: false
   gem 'heroku_rails_deploy', '>= 0.4.1', require: false
   gem 'overcommit'
-  gem 'spring'
```
Comment (erikkessler1) [inline Gemfile:34]: I removed Spring as it is no longer recommended for Rails: https://github.com/rails/rails/pull/42997

**Review by atsheehan — APPROVED** [review]
Looks good to me. Have you tested the Zeitwerk upgrade in staging?

File: app/models/schemas/register_new_version.rb [hunk 2]
```diff
@@ -8,14 +8,16 @@ module Schemas
   # created models.
   # If a unique index error is raised, then the operation is retried once.
   class RegisterNewVersion
-    include Procto.call
```
[the remark this answers is not in the document]
Comment (erikkessler1) [reply app/models/schemas/register_new_version.rb:11]: Yeah it hasn't been touched in years from what I can tell and leads to keyword arg issues with newer versions of Ruby due to how it delegates arguments.

**Comment by erikkessler1** [comment]
> Looks good to me. Have you tested the Zeitwerk upgrade in staging?

Yeah I tested both reading from and writing to the registry in staging.

**Pull request merged**